node-services/common/src/common/server-side/logger.ts

```diff
@@ -11,54 +11,74 @@ import winston, {format} from 'winston';
 import DailyRotateFile from 'winston-daily-rotate-file';
 import config from 'config';
 
-const logFormat = winston.format.combine(
-    winston.format.colorize(),
-    winston.format.timestamp(),
-    winston.format.align(),
-    winston.format.printf((info) => `${info.timestamp} ${info.level}: ${info.message}`)
-);
+export default class Logger {
+    static logger: winston.Logger;
 
-const logConfiguration: any = config.get('operatorfabric.logConfig');
+    static logFormat = winston.format.combine(
+        winston.format.colorize(),
+        winston.format.timestamp(),
+        winston.format.align(),
+        winston.format.printf((info) => `${info.timestamp} ${info.level}: ${info.message}`)
+    );
 
-let logger = winston.createLogger({
-    format: format.combine(
-        format((info: winston.LogEntry) => {
-            info.message = prefixNewLinesToAvoidLogInjection(info.message);
-            return info;
-        })(),
-        logFormat
-    ),
-    transports: [new winston.transports.Console({level: logConfiguration.logLevel})]
-});
+    static logConfiguration: any = config.get('operatorfabric.logConfig');
 
-if (logConfiguration.logFile != null) {
-    const transport = new DailyRotateFile({
-        filename: logConfiguration.logFolder + logConfiguration.logFile,
-        datePattern: 'YYYY-MM-DD',
-        zippedArchive: true,
-        maxSize: '20m',
-        maxFiles: '14d',
-        level: logConfiguration.logLevel
-    });
+    public static getLogger(): winston.Logger {
+        this.logger = winston.createLogger({
+            format: format.combine(
+                format((info) => {
```

Comment:
I suggest to create the logger only if it does not exist (i.e only once)

Even if in the actual code , the service call only one time getLogger() , it may be use differently by a developer in the future . (for example getting logger in each class instead of passing the logger object) .